docs/apache-airflow-providers-amazon/example-dags.rst

```diff
@@ -20,4 +20,4 @@ Example DAGs
 
 You can learn how to use Amazon AWS integrations by analyzing the source code of the example DAGs:
 
-* `Amazon AWS <https://github.com/apache/airflow/tree/main/airflow/providers/amazon/aws/example_dags>`__
+* `Amazon AWS <https://github.com/apache/airflow/tree/providers-amazon/4.0.0/tests/system/providers/amazon/aws>`__
```

Comment:
For the providers that are still in progress for their transition from example DAGs to system tests (e.g. Amazon, Google), should we default to pointing to system tests even if there are still example DAGs in the original directory (and in the case of Amazon, the vast majority haven't been migrated)? Maybe we check if the example DAG path exists first in the pre-commit check instead?

Reply:
Well it's wrong both ways in fact ... But I could modify it slightly to add BOTH links in this case :) Hold my beer.

Reply:
Should be fine now :) . We should have legacy links now for both google and amazon (those are the only ones which  are in-progress.. And the legacy link will disappear when completed.

While doing so, I also fixed an inconsistency in google examples - the "cloud" examples weren't in "cloud" subfolder in tests/systems/. This is now fixed and verified by pre-commit too, so it was worth the effort :)